pages/deliverer/incoming-orders/[id].tsx

```diff
@@ -0,0 +1,201 @@
+  name: string
+}
+
+const IncomingOrder = ({ user }) => {
+  const router = useRouter()
+  const [alert, setAlert] = useAlert()
+
+  // will be passed in from the homescreen, when the user selects the incoming order
+  const { id } = router.query
+  const orderId: number = Number(id)
+
+  const [order, setOrder]: [
+    IncomingOrderDto,
+    Dispatch<SetStateAction<IncomingOrderDto>>,
+  ] = useState()
+
+  // simulate api call, pass in props.id
+  useEffect(() => {
+    const query = async () => {
+      const { data: ordersData, error: ordersError } = await supabaseClient
+        .from<IncomingOrderDto>('orders')
+        .select(
+          'id, restaurants (id, name), order_text, ordered_at, is_active, destinations (id, name), cost',
+        )
+        .eq('id', orderId)
+        .is('deliverer_id', null)
+        .limit(1)
+        .single()
+
+      if (ordersError) {
+        router.push('/home')
+        setAlert({
+          type: 'warning',
+          message: ordersError.code + ': ' + ordersError.message,
+        })
+        console.log(ordersError)
+        return
+      }
+
+      // means a deliverer has already takent the order
+      if (ordersData.deliverer_id) {
+        router.push('/home')
+        setAlert({ type: 'info', message: 'Order has been taken' })
+        return
+      }
+
+      console.log(ordersData)
+
+      setOrder({ ...ordersData, ordered_at: new Date(ordersData.ordered_at) })
+    }
+
+    query()
+
+    //not best practice, need to figure out how to handle this
+  }, [])
+
+  async function handleReject() {
+    console.log(
+      'store in db rejected, never show this order on the homescreen again',
+    )
+    let error = ''
+    const { error: removePendingError } = await supabaseClient.rpc(
+      'remove_pending_order',
+      { order_id: id, user_id: user.id },
+    )
+    if (removePendingError) {
+      error +=
+        removePendingError.code + ': ' + removePendingError.message + '\n'
+    }
+    const { error: addRejectedError } = await supabaseClient.rpc(
+      'add_rejected_user',
+      { order_id: id, user_id: user.id },
+    )
+    if (addRejectedError) {
+      error += addRejectedError.code + ': ' + addRejectedError.message
+    }
+
+    if (error) {
+      setAlert({ type: 'warning', message: error })
+    } else {
+      setAlert({ type: 'info', message: 'Rejected order' })
+    }
+    router.push('/home')
+  }
+
+  async function handleAccept() {
+    // implement auth to get the user uid
+    const { data: updatedOrdersData, error: updatedOrdersError } =
+      await supabaseClient
+        .from('orders')
+        .update({ id: id, deliverer_id: user.id })
+        .eq('id', id)
+        .is('is_active', true)
+        .is('deliverer_id', null)
+
+    // possible that order has been taken or completed
+    if (updatedOrdersError) {
+      const { data: ordersData, error: ordersError } = await supabaseClient
+        .from<IncomingOrderDto>('orders')
+        .select(
```

Comment:
Is there a need to select all these properties? if the idea here is that we want to check if the order has been accepted, just selecting the deliverer is ok?